drivers/sensor/Kconfig.sensor_clock

```diff
@@ -0,0 +1,30 @@
+# Sensor clock configuration options
+# Copyright(c) 2024 Cienet
+# SPDX-License-Identifier: Apache-2.0
+
+config SENSOR_CLOCK
+	bool
+	default y if SENSOR_ASYNC_API
+	help
+	  Configure the sensor clock source for the system.
+
+if SENSOR_CLOCK
+
+choice
+	prompt "Sensor clock type"
+	default SENSOR_CLOCK_SYSTEM
+	help
+	  Select the clock source to be used for sensor timing.
+
+config SENSOR_CLOCK_SYSTEM
+	bool "Use the system counter for sensor time"
+
+config SENSOR_CLOCK_COUNTER
+	bool "Use a counter device/API for sensor time"
+
+config SENSOR_CLOCK_RTC
+	bool "Use an RTC device/API for sensor time"
+
+endchoice
```

Comment:
I was going through #63651 and it seems this Kconfig choice limits our ability to have different clocks per sensor, mentioned in: https://github.com/zephyrproject-rtos/zephyr/issues/63651#issuecomment-1765634695.

Isn't it better to have it implemented as a DT-node property that then is handled on a per-driver basis?

Reply:
I should have commented on that thread again. I'm on the fence. This (with the kconfig) allows us to do tick based math on time across multiple sensors if needed. I'm happy with either approach with a slight bias to keep things simple with a single sensor clock with I think will cover the vast majority of use cases. Having 2 clocks for sensors comes with it's own problems that I ran into when running some tests in a sandbox (since the edges don't always align it's very hard to do any form of fusion correctly).